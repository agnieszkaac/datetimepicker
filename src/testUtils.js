Date.now = jest.fn(() => Date.parse("2020-01-01"));
