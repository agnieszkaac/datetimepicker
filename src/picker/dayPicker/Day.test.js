import React from "react";
import moment from "moment";
import { render, fireEvent } from "@testing-library/react";

import { Day } from "./";
import { setToday } from "../../testUtils";

describe("<Day /> should", () => {
  setToday();

  let value = null;
  const displayValue = "1";

  //default props
  const date = moment();
  const onClick = jest.fn((e) => (value = e.target.value));
  const props = {
    date,
    selected: false,
    className: "test",
    onClick,
  };

  it("match snapshot", () => {
    const { asFragment } = render(<Day {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("have correct display value for default date", () => {
    const { getByTestId } = render(<Day {...props} />);
    const day = getByTestId("day");

    expect(day).toHaveTextContent(displayValue);
  });

  it("have selected class when selected", () => {
    const { getByTestId, rerender } = render(<Day {...props} />);
    const day = getByTestId("day");

    expect(day).not.toHaveClass("selected");

    rerender(<Day {...props} selected={true} />);
    expect(day).toHaveClass("selected");
  });

  it("fire onClick with proper value when clicked", () => {
    const { getByTestId } = render(<Day {...props} />);
    const day = getByTestId("day");
    const expectedValue = date.toISOString();

    expect(day).toHaveValue(expectedValue);

    fireEvent.click(day);

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(value).toBe(expectedValue);
  });
});
