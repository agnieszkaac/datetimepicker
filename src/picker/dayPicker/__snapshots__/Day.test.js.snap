// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> should match snapshot for NOT present day and NOT selected 1`] = `
<div>
  <button
    class="test day"
    type="button"
    value="2020-01-02T00:00:00.000Z"
  >
    2
  </button>
</div>
`;

exports[`<Day /> should match snapshot for NOT present day and selected 1`] = `
<div>
  <button
    class="test day selected"
    type="button"
    value="2020-01-02T00:00:00.000Z"
  >
    2
  </button>
</div>
`;

exports[`<Day /> should match snapshot for present day and NOT selected 1`] = `
<div>
  <button
    class="test day is-today"
    type="button"
    value="2020-01-01T00:00:00.000Z"
  >
    1
  </button>
</div>
`;

exports[`<Day /> should match snapshot for present day and selected 1`] = `
<div>
  <button
    class="test day selected is-today"
    type="button"
    value="2020-01-01T00:00:00.000Z"
  >
    1
  </button>
</div>
`;
